### ISSUE TITLE
ImplicitAnimations don't work for the first three changes

### ISSUE THREAD

**Issue by laolarou726** [AvaloniaUI/Avalonia#15254]
### Describe the bug

ImplicitAnimations don't work for the first three changes.
Furthermore, when setting the animation key for the ImplicitAnimations, if the key name does not equal the property that is going to animate. The ImplicitAnimations will not work at all.

For example:

```c#
var ani = compositor.CreateVector3KeyFrameAnimation();

ani.Target = "Offset";
ani.InsertExpressionKeyFrame(1f, "this.FinalValue", easing);
ani.Duration = TimeSpan.FromMilliseconds(445);

visual.ImplicitAnimations ??= compositor.CreateImplicitAnimationCollection();

visual.ImplicitAnimations["Offset"] = ani; // animation will work but still buggy
visual.ImplicitAnimations["A_RANDOM_NAME"] = ani; // not working at all
```

![gif](https://github.com/AvaloniaUI/Avalonia/assets/25716486/1635c5be-d57e-436c-9dc2-7c37a9d96f2c)

### To Reproduce

See repro: [https://github.com/laolarou726/CompositionBug](https://github.com/laolarou726/CompositionBug)
Also see issue: #15096 

### Expected behavior

All the animations should work for every value change and trigger.

### Avalonia version

latest nightly

### OS

Windows

**Comment by lindexi** [comment]
@laolarou726 Your https://github.com/laolarou726/CompositionBug is 404

**Comment by laolarou726** [comment]
> @laolarou726 Your https://github.com/laolarou726/CompositionBug is 404

Sorry, that was a mistake. Now restored!

**Comment by lindexi** [comment]
Thank you @laolarou726 

I fixed your issues in https://github.com/AvaloniaUI/Avalonia/pull/17370

**Comment by laolarou726** [comment]
> Thank you @laolarou726
> 
> I fixed your issues in #17370

LGTM! Thank you so much for your help on this issue!
